packages/ember-application/tests/system/reset_test.js

```diff
@@ -1,234 +1,142 @@
 
-    application.reset();
+    this.application.reset();
 
-    equal(eventDispatcherWasDestroyed, 1);
-    equal(eventDispatcherWasSetup, 2, 'setup called after reset');
-  } catch (error) { Registry.prototype.register = originalRegister; }
-
-  Registry.prototype.register = originalRegister;
-});
+    assert.equal(eventDispatcherWasDestroyed, 1);
+    assert.equal(eventDispatcherWasSetup, 2, 'setup called after reset');
+  }
 
-QUnit.test('When an application is reset, the router URL is reset to `/`', function() {
-  let location, router;
+  ['@test When an application is reset, the router URL is reset to `/`'](assert) {
+    run(() => {
+      this.createApplication();
 
-  run(() => {
-    application = Application.create();
-    application.Router = Router.extend({
-      location: 'none'
-    });
+      this.add('router:main', Router.extend({
+        location: 'none'
+      }));
 
-    application.Router.map(function() {
-      this.route('one');
-      this.route('two');
+      this.router.map(function() {
+        this.route('one');
+        this.route('two');
+      });
     });
-  });
-
-  router = application.__container__.lookup('router:main');
-
-  location = router.get('location');
-
-  run(() => {
-    location.handleURL('/one');
-  });
-
-  application.reset();
-
-  let applicationController = application.__container__.lookup('controller:application');
-  router = application.__container__.lookup('router:main');
-  location = router.get('location');
-
-  equal(location.getURL(), '');
-
-  equal(get(applicationController, 'currentPath'), 'index');
-
-  location = application.__container__.lookup('router:main').get('location');
-  run(() => {
-    location.handleURL('/one');
-  });
-
-  equal(get(applicationController, 'currentPath'), 'one');
-});
 
-QUnit.test('When an application with advance/deferReadiness is reset, the app does correctly become ready after reset', function() {
-  var readyCallCount;
+    this.visit('/one');
 
-  readyCallCount = 0;
+    this.application.reset();
 
-  run(() => {
-    application = Application.create({
-      ready() {
-        readyCallCount++;
-      }
-    });
-
-    application.deferReadiness();
-    equal(readyCallCount, 0, 'ready has not yet been called');
-  });
+    let applicationController = this.applicationInstance.lookup('controller:application');
+    let router = this.applicationInstance.lookup('router:main');
+    let location = router.get('location');
 
-  run(() => {
-    application.advanceReadiness();
-  });
+    assert.equal(location.getURL(), '');
+    assert.equal(get(applicationController, 'currentPath'), 'index');
 
-  equal(readyCallCount, 1, 'ready was called once');
+    this.visit('/one');
 
-  application.reset();
-
-  equal(readyCallCount, 2, 'ready was called twice');
-});
+    assert.equal(get(applicationController, 'currentPath'), 'one');
+  }
 
-QUnit.test('With ember-data like initializer and constant', function() {
```

Comment:
This test is simply removed. `DS.defaultStore` is no longer a thing. This test asserts that `init` is called on the registered factory, and that `willDestroy` is not called when `reset` is called without a runloop. This is not meaningful.